### ISSUE TITLE
Not being able to set the `into` argument of `@separate`

### ISSUE THREAD

**Issue by atantos** [TidierOrg/TidierData.jl#35]
Suppose I have a DataFrame column named `:embeddings`. Each row in this column contains an array of floats. Since the function `unnest_wider()` isn't available yet, I'm considering an alternative approach. Since, `unnest_wider()` is not available yet, I thought I could first concatenate the array elements into a single string. After that, I planned to use the @separate macro to split the string, effectively emulating the `unnest_wider()` function.

However, I've encountered an issue. When I try to specify the `into` argument with `new_names`, I receive an error: `"ERROR: LoadError: MethodError: no method matching iterate(::QuoteNode)"`. I've reviewed the macro's source code and didn't find any apparent issues. Interestingly, if I omit the new_names argument (similar to R's `separate()` function where it auto-generates column names), I get another error: `"ERROR: LoadError: MethodError: no method matching var" @separate"(::LineNumberNode, ::Module, ::Symbol, ::Symbol, ::String)"`.

```julia
new_names = ["x$(i)" for i in 1:1536];
println("sd")
horror_movies_tidier_ =
    @chain horror_movies_tidier begin
        transform(_, :embeddings => ByRow(x -> join(x, "_")) => :embeddings)
    end
@chain horror_movies_tidier_ begin
    @separate(embeddings, new_names, "_")
end
```

**Comment by kdpsingh** [comment]
We'll take a look. I think what's happening here is that `@separate()` is assuming that `new_names` refers to the actual name of the column to be created rather than a container of column names.

Normally, we would handle this with interpolation but `@separate()`, as currently implemented, doesn't support interpolation yet.

This should be fixable. Stay tuned.

**Comment by atantos** [comment]
Looking forward to the fix!

**Comment by drizk1** [comment]
I think I have fixed this. I'll put in the PR soon.

**Comment by kdpsingh** [comment]
Fixed in #40 and merged.

From your example, the following syntax should work now:

```julia
@chain horror_movies_tidier_ begin
    @separate(embeddings, !!new_names, "_")
end
```

Prefixing a `!!` to `new_names` tells TidierData.jl to look in the parent environment for `new_names` because the default assumption in TidierData.jl is that all "bare" symbols refer to column names.

**Comment by kdpsingh** [comment]
Thanks to @drizk1 for all the hard work on this.
